Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Templates/APIGatewaySetupParameters.tt

```diff
@@ -244,6 +244,57 @@
                 validationErrors.Add($"Value {__request__.Body} at 'body' failed to satisfy constraint: {e.Message}");
             }
 
+<#
+                }
+            }
+            else if (parameter.Attributes.Any(att => att.Type.FullName == TypeFullNames.FromCustomAuthorizerAttribute))
+            {
+                var fromAuthorizerAttribute = parameter.Attributes?.FirstOrDefault(att => att.Type.FullName == TypeFullNames.FromCustomAuthorizerAttribute) as AttributeModel<Amazon.Lambda.Annotations.APIGateway.FromCustomAuthorizerAttribute>;
+
+                // Use parameter name as key, if Name has not specified explicitly in the attribute definition.
+                var authKey = fromAuthorizerAttribute?.Data?.Name ?? parameter.Name;
+                if(restApiAttribute != null)
+                {
+#>
+            var <#= parameter.Name #> = default(<#= parameter.Type.FullName #>);
+            if (__request__.RequestContext?.Authorizer == null)
+            {
+                validationErrors.Add("Could not find Authorizer data for request");
```

Comment:
I'm wondering if there is missing data from the authorizer if we should be returning back `401` unauthorized instead of a bad parameter status code.

Reply:
Good point. Away from pc for 2 weeks.
Feel free to change or I will once I am  back at desk.

Reply:
Good point. It should return 401.
I am away from pc for 2 weeks, feel free to change or i will once back.

Reply:
Would you be able to have a look at Norm's comments?